pil_analyzer/src/pil_analyzer.rs

```diff
@@ -570,23 +570,13 @@ impl<T: FieldElement> PILAnalyzer<T> {
                 let body = Box::new(self.process_function(&params, *body));
                 Expression::LambdaExpression(LambdaExpression { params, body })
             }
-            PExpression::BinaryOperation(left, op, right) => {
-                if let Some(value) = self.evaluate_binary_operation(&left, op, &right) {
```

Comment:
This used to evaluate expressions on the parsed AST, which is weird and blocks a proper stage-based approach because we can only properly evaluate expressions once we have types, and we will not have types at this stage yet.